### PR TITLE
Adding macOS CI

### PR DESCRIPTION
**Pull request by henryiii**
This adds macOS support to the ROOT package. General changes that affect Linux and macOS:

* Added Pythia8
* ROOTSYS is set in all shells instead of just bash
* `CMAKE_BUILD_WITH_INSTALL_RPATH` patch removed
* `LIBXML2_INCLUDE_DIRS` fix (merged in ROOT)
* Remove `SetLibraryPath` from ROOT executable
* Remove ROOT's custom `FindGSL.cmake` (ROOT will do this as well soon)
* Explicit `DCMAKE_INSTALL_NAME_DIR` and `PYTHON_EXECUTABLE `
* Removed `gvis` as it's being dropped in ROOT 6.18
* Download release instead of checkout git (much smaller/faster for CI)
* Clang-patches can be controlled separately
* Since ROOT uses Numpy in the build, use pin_compatible numpy
* Add a couple of tests
* Set ROOTSYS in a few more shells
* Afterimage feedstock comes from ROOT source now

macOS specific features:

* Target `MACOSX_DEPLOYMENT_TARGET` explicitly
* CLANG_RESOURCE_DIR_VERSION (merged in ROOT 6.16.02)
* Patch ROOT to not link to Python directly, special check for PyROOT and PyMVA from ROOT
* Filter out the capture of an absolute file path
* Add warning about SDK in bash and fish
* Also set SDKROOT (might fix Azure)

### CODE REVIEW

**Comment by henryiii** [comment]
@conda-forge-admin, please rerender

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by chrisburr** [comment]
Before I forget, it might be easier to disable cocoa support for now as I think that is the issue here.

**Comment by henryiii** [comment]
I finally have this compiling locally. With no changes, I get an error at about 60% with cling (didn't copy it). If I have ROOT build clang, then I get to here:

```
[100%] Generating etc/allDict.cxx.pch
/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_build_env/bin/cmake -E env ROOTIGNOREPREFIX=1 /Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/python /Users/henryiii/anaconda3/conda-bld/root_1549019432410/work/root-source/etc/dictpch/makepch.py etc/allDict.cxx.pch -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/work/build-dir/include
./bin/rootcling -rootbuild -1 -f allDict.cxx -noDictSelection -c -D__CLING__ -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DROOT_PCH -fsigned-char -I./include -I./etc -I./etc/dictpch -I./etc/cling -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/freetype2 -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/graphviz -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/libxml2 -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/python3.7m -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/xrootd -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/xrootd/private -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/numpy/core/include -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -pipe -stdlib=libc++ -fvisibility-inlines-hidden  -fmessage-length=0 -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include -fdebug-prefix-map=/Users/henryiii/anaconda3/conda-bld/root_1549019432410/work=/usr/local/src/conda/root-6.16.00 -fdebug-prefix-map=/Users/henryiii/anaconda3/conda-bld/root_1549019432410/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla=/usr/local/src/conda-prefix -isysroot /opt/MacOSX10.9.sdk -m64 -pipe -fsigned-char -fno-common -Qunused-arguments -pthread -std=c++1z -stdlib=libc++ -DSYSTEM_TYPE_unix -I/Users/henryiii/anaconda3/conda-bld/root_1549019432410/work/build-dir/include etc/dictpch/allHeaders.h  etc/dictpch/allLinkDefs.h
dyld: Library not loaded: @rpath/libRIO.6.16.so
  Referenced from: /Users/henryiii/anaconda3/conda-bld/root_1549019432410/work/build-dir/./bin/rootcling
  Reason: image not found
make[2]: *** [etc/allDict.cxx.pch] Error 250
make[1]: *** [CMakeFiles/onepcm.dir/all] Error 2
```

(this is only on my local computer at the moment, so homebrew exists too. I'll hopefully be able to set this back up on my spare computer next week)

**Comment by henryiii** [comment]
Note: I'm dropping `gvis` since ROOT is printing a warning saying it will be removed in ROOT 6.18.

**Comment by henryiii** [comment]
Hey, I've actually got the same error locally as in the online build! Not sure why it popped up now, but it's easier to debug. Maybe the libcxx=4 addition?

```
CMake Error at /Users/henryiii/anaconda3/conda-bld/root_1549125673984/_build_env/lib/cmake/llvm/LLVMExports.cmake:1167 (message):
  The imported target "PollyISL" references the file

     "/Users/henryiii/anaconda3/conda-bld/root_1549125673984/_build_env/lib/libPollyISL.a"

  but this file does not exist.  Possible reasons include...
```

**Comment by chrisburr** [comment]
Yeah `libcxx=4` probably shouldn't be part of the host dependencies

This means ROOT is finding `llvm-config` from llvm 4 (default macOS compiler) instead of llvm 5 (host dependency).

**Comment by henryiii** [comment]
Here's the error when it does not try to build CLING.


```
Dependee "/Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/core/metacling/src/CMakeFiles/MetaCling.dir/DependInfo.cmake" is newer than depender "/Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/core/metacling/src/CMakeFiles/MetaCling.dir/depend.internal".
Dependee "/Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/core/metacling/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/core/metacling/src/CMakeFiles/MetaCling.dir/depend.internal".
Scanning dependencies of target MetaCling
/Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/root-source/core/clingutils/src/TClingUtils.cxx:3281:33: error: too many arguments to function call, expected at most 13, have 15
                                false /*OpenFile*/, true /*CacheFailures*/);
                                ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/henryiii/anaconda3/conda-bld/root_1549127832594/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/clang/Lex/HeaderSearch.h:381:3: note: 'LookupFile' declared here
  const FileEntry *LookupFile(
  ^
-- clad configure command succeeded.  See also /Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/interpreter/cling/tools/plugins/clad/clad-prefix/src/clad-stamp/clad-configure-*.log
cd /Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/interpreter/cling/tools/plugins/clad/clad-prefix/src/clad-build && /Users/henryiii/anaconda3/conda-bld/root_1549127832594/_build_env/bin/cmake -E touch /Users/henryiii/anaconda3/conda-bld/root_1549127832594/work/build-dir/interpreter/cling/tools/plugins/clad/clad-prefix/src/clad-stamp/clad-configure
```

**Comment by chrisburr** [comment]
As mattermost is down, I have another broken test to add:
```bash
python -c 'import ROOT; ROOT.TBrowser()'
```
It has nothing to do with Python and I can trigger it interactively using the root prompt. Using `root -e "new TBrowser()"` exits before it segfaults. It's caused by the `libAfterImage.so.0` error that is already shown in the tests.

**Comment by chrisburr** [comment]
To fix Azure this needs to be removed (or patched to copy from `${PREFIX}/include/c++/v1/wchar.h`:

https://github.com/root-project/root/blob/d52847c4ce6162ac6945eff7575bdf8fcf1c915c/core/clingutils/CMakeLists.txt#L122-L131

File: recipe/meta.yaml [hunk 1]
```diff
@@ -131,6 +138,8 @@ test:
     - JupyROOT
   commands:
     - ipython -c 'import JsMVA'
+    - root.exe -b -l -q -e
```
Comment (chrisburr) [inline recipe/meta.yaml]: I think these two tests can be removed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -93,17 +99,18 @@ requirements:
     - graphviz
     - gsl
     - krb5  # [osx]
+    - libcxx=4  # [osx]
```
Comment (chrisburr) [inline recipe/meta.yaml]: This can be dropped now the compilers are a runtime dependency

**Comment by henryiii** [comment]
Tomorrow, if the tests pass, I think it will be ready for a squash and merge!

**Comment by chrisburr** [comment]
Here we go! 🤞

**Pull request merged**